tools/osx/crosstool/BUILD.toolchains

```diff
@@ -23,6 +23,14 @@ OSX_TOOLS_CONSTRAINTS = {
         "@platforms//os:osx",
         "@platforms//cpu:x86_64",
     ],
+    "maccatalyst_x86_64": [
+        "@platforms//os:ios",
+        "@platforms//cpu:x86_64",
+    ],
+    "maccatalyst_arm64": [
+        "@platforms//os:ios",
```

Comment:
maybe we should use `macOS` here? or defining new platform for Catalyst?

Not sure about this.

Reply:
So here's where we might have a problem. The ideal solution here is that we add a third entry to this list to disambiguate this, but the value we need to add is in the apple_support repo, which bazel does not currently depend on. I've been lightly talking with @gregestren about what to do for this case, but we haven't reached a conclusion.

Reply:
But in the meantime I'm not sure if this should be ios or macOS. Probably macOS given that's what it's targeting running on?

Reply:
I think `@platforms//os:osx` (macOS) makes sense! Because when we're using `xcodebuild` to compile an iOS project for Catalyst, the value of `-sdk` flag is `macosx`.

Reply:
gentle ping @gregestren 

do we want to move forward on Mac Catalyst support? 
I can take the action when you have any feedback.

Reply:
I expect to discuss the bazel -> apple_support dependency at a meeting at the end of October. I don't have strong opinions in the meantime - it reads like this conversation has an interim consensus?

Reply:
@gregestren Thanks for your reply! No pressure on this. Because I'm in a self doubt on this. I can hardly see any interest about community support of Catalyst. This is not only limited to Bazel. While using Rust, I'm facing the same issue.

I'm thinking about writing macOS app instead of a Catalyst app.